nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -111,93 +91,54 @@ public MongoDBDirectory getDirectory() {
     }
 
     @Override
-    public DocumentModel getEntry(String id) throws DirectoryException {
-        return getEntry(id, true);
-    }
-
-    @Override
-    public DocumentModel getEntry(String id, boolean fetchReferences) throws DirectoryException {
-        if (!hasPermission(SecurityConstants.READ)) {
-            return null;
-        }
-        return directory.getCache().getEntry(id, this, fetchReferences);
-    }
-
-    @Override
-    public DocumentModelList getEntries() throws DirectoryException {
-        if (!hasPermission(SecurityConstants.READ)) {
-            return new DocumentModelListImpl();
-        }
-        return query(Collections.emptyMap());
-    }
-
-    @Override
-    public DocumentModel createEntry(Map<String, Object> fieldMap) throws DirectoryException {
-        checkPermission(SecurityConstants.WRITE);
+    protected DocumentModel createEntryWithoutReferences(Map<String, Object> fieldMap) {
+        // Filter out reference fields for creation as we keep it in a different collection
+        Map<String, Object> newDocMap = fieldMap.entrySet()
+                                                .stream()
+                                                .filter(entry -> getDirectory().getReferences(entry.getKey()) == null)
+                                                .collect(HashMap::new, (m, v) -> m.put(v.getKey(), v.getValue()),
+                                                        HashMap::putAll);
```

Comment:
It happens that the `fieldMap` contains null values, and it appears that `Collectors.toMap()` doesn't handle null values. (cf https://stackoverflow.com/questions/24630963/java-8-nullpointerexception-in-collectors-tomap)